splitgraph/ingestion/sqlite/__init__.py

```diff
@@ -0,0 +1,177 @@
+import contextlib
+import itertools
+import os
+import sqlite3
+import tempfile
+from collections.abc import Generator
+from contextlib import contextmanager
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, cast
+
+import requests
+from psycopg2.sql import SQL, Identifier
+
+from splitgraph.core.types import (
+    Credentials,
+    IntrospectionResult,
+    Params,
+    TableColumn,
+    TableInfo,
+    TableParams,
+)
+from splitgraph.hooks.data_source.base import LoadableDataSource
+
+if TYPE_CHECKING:
+    from splitgraph.engine.postgres.engine import PostgresEngine
+
+LIST_TABLES_QUERY = """
+SELECT * FROM
+(SELECT
+	tbl_name
+FROM sqlite_master
+WHERE type='table') t
+JOIN pragma_table_info(tbl_name) s
+ORDER BY 1,2;
+"""
+
+# based on https://stackoverflow.com/a/16696317
+def download_file(url: str, local_fh: tempfile._TemporaryFileWrapper) -> int:
+    total_bytes_written = 0
+    with requests.get(url, stream=True, verify=os.environ["SSL_CERT_FILE"]) as r:
+        r.raise_for_status()
+        for chunk in r.iter_content(chunk_size=8192):
+            total_bytes_written += local_fh.write(chunk)
+    return total_bytes_written
+
+
+@contextmanager
+def minio_file(url: str) -> Generator[str, None, None]:
+    with tempfile.NamedTemporaryFile(mode="wb", delete=False) as local_fh:
+        download_file(url, local_fh)
+    yield local_fh.name
+    os.remove(local_fh.name)
```

Comment:
The os.remove should either be in try..finally to make sure we don't keep that file on disk in case of a crash, or I think you could do this and it'll delete the file when _your_ context manager is exited?

```suggestion
    with tempfile.NamedTemporaryFile(mode="wb", delete=True) as local_fh:
        download_file(url, local_fh)
        yield local_fh.name
```

Reply:
The issue was that until the file is closed, sqlite won't be able to read it, reporting the file as corrupt. I now realize that calling `flush()` is enough, allowing me to use the more elegant `delete=True` solution. Thanks!